File: Kuroba/app/src/main/java/com/github/adamantcheese/chan/ui/controller/settings/MediaSettingsControllerCallbacks.java
package com.github.adamantcheese.chan.ui.controller.settings;

import androidx.annotation.Nullable;

import com.github.k1rakishou.fsaf.file.AbstractFile;

public interface MediaSettingsControllerCallbacks {
    void updateLocalThreadsLocation(String newLocation);

    void askUserIfTheyWantToMoveOldThreadsToTheNewDirectory(
            @Nullable AbstractFile oldBaseDirectory, AbstractFile newBaseDirectory
    );

    void askUserIfTheyWantToMoveOldSavedFilesToTheNewDirectory(
            @Nullable AbstractFile oldBaseDirectory, AbstractFile newBaseDirectory
    );

    void updateLoadingViewText(String text);

    void updateSaveLocationViewText(String newLocation);

    void showCopyFilesDialog(int filesCount, AbstractFile oldBaseDirectory, AbstractFile newBaseDirectory);

    void onCopyDirectoryEnded(AbstractFile oldBaseDirectory, AbstractFile newBaseDirectory, boolean result);
}

File: Kuroba/app/src/main/java/com/github/adamantcheese/chan/ui/controller/MediaSettingsControllerCallbacks.kt
package com.github.adamantcheese.chan.ui.controller

import android.widget.Toast
import com.github.k1rakishou.fsaf.file.AbstractFile

interface MediaSettingsControllerCallbacks {
    fun showToast(message: String, length: Int = Toast.LENGTH_SHORT)
    fun updateLocalThreadsLocation(newLocation: String)

    fun askUserIfTheyWantToMoveOldThreadsToTheNewDirectory(
            oldBaseDirectory: AbstractFile,
            newBaseDirectory: AbstractFile
    )

    fun askUserIfTheyWantToMoveOldSavedFilesToTheNewDirectory(
            oldBaseDirectory: AbstractFile,
            newBaseDirectory: AbstractFile
    )

    fun updateLoadingViewText(text: String)
    fun updateSaveLocationViewText(newLocation: String)

    fun showCopyFilesDialog(
            filesCount: Int,
            oldBaseDirectory: AbstractFile,
            newBaseDirectory: AbstractFile
    )

    fun onCopyDirectoryEnded(
            oldBaseDirectory: AbstractFile,
            newBaseDirectory: AbstractFile,
            result: Boolean
    )
}